### ISSUE TITLE
[Modern] Edge/Node objects from PaginationContainer are not equal after each pagination

### ISSUE THREAD

**Issue by andyentity** [facebook/relay#2322]
The objects representing the edges/nodes returned by the `PaginationContainer` are not equal after each pagination step, i.e. `a !== b` when `const a = this.props.connectionObjs.edges[0]` before a pagination step and `const b = this.props.connectionObjs.edges[0]` after the pagination step.

One would expect that the reference of a edge/node object that has already been retrieved does not change after a pagination step and new node objects are really just appended to the array, however this is not the case.
Now, if I commit a mutation that changes nodes in that list I cannot simply afterwards check by equality operator `===` if the node changed because it always returns `false` even when there are no changes, instead I needs to perform a deep equality check.

## Example:

Using Relay Modern 1.4.1.

This is a minimal example of a table that gets the first 10 nodes from the `QueryRenderer` (not shown) and retrieves the next node in the connection via the `PaginationContainer`.

Note that in the `render()` function I added a line that assigns the array of edges that have been retrieved up to now to a global variable in order to demonstrate the issue.

```
import React from 'react';
import { createPaginationContainer, graphql } from 'react-relay';

class Table extends React.Component {
  static get COLUMNS() {
    return ['firstname', 'lastname', 'city', 'zip_code', 'street', 'house_number'];
  }

  headers() {
    return <tr>
      {Table.COLUMNS.map(col => <th key={col}>{col}</th>)}
    </tr>;
  }

  row(user) {
    return <tr key={user.id}>
      {Table.COLUMNS.map(col => <td key={col}>{user[col]}</td>)}
    </tr>;
  }

  loadMore() {
    if(!this.props.relay.hasMore()) {
      alert("No more items available!")
      return;
    } else if(this.props.relay.isLoading()) {
      alert("Already loading... please wait")
      return;
    }

    this.props.relay.loadMore(1);
  }

  render() {
    if(this.props.users && this.props.users.paginateUsers) {
      window.asdf = this.props.users.paginateUsers;
    };
    return [
      <table key="tbl">
        <tbody>
          {this.headers()}
          {this.props.users.paginateUsers.edges.map(e => this.row(e.node))}
        </tbody>
      </table>,
      <button key="btn" onClick={() => this.loadMore()}>Load More</button>
    ];
  }
}

export default createPaginationContainer(Table, graphql`
  fragment Table_users on BackofficeQuery {
    paginateUsers(
      first: $count,
      after: $after
    ) @connection(key: "Table_paginateUsers") {
      edges {
        node {
          id
          firstname
          lastname
          city
          zip_code
          street
          house_number
        }
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
`, {
  direction: 'forward',
  query: graphql`
    query TableForwardQuery(
      $count: Int!,
      $after: String,
    ) {
      ...Table_users
    }
  `,
  getConnectionFromProps(props) {
    return props.users && props.users.paginateUsers;
  },
  getFragmentVariables(previousVariables, totalCount) {
    return {
      ...previousVariables,
      count: totalCount,
    };
  },
  getVariables(props, paginationInfo, fragmentVariables) {
    return {
      count: paginationInfo.count,
      after: paginationInfo.cursor,
    };
  },
})
```

Now if you do `window.jkl = window.asdf.edges[0]` and click on the button to paginate, after the second pagination step onwards `window.jkl === window.asdf.edges[0]` returns `false` where `true` is expected.

**Comment by sibelius** [comment]
Migrate to relay hooks, if the issue still valid, open another issue
